### DISCUSSION TITLE
What BIDS name do we want for our derivatives ?

### DISCUSSION THREAD

**Discussion by Remi-Gau** [cpp-lln-lab/bidspm#558] [General]
As discussed starting a discussion thread on how files will be renamed into BIDS derivatives.

One thing to clarify: the final output of a preprocessing pipeline (for func or anat) MUST be described as `desc-preproc`. This is done to facilitate knowing which file to "grab" no matter what software is used afterwards.

Also "provenance tracking" knowing the "parent" files of given file can be done in BIDS derivatives in the side-car json files, so it might not have to be encoded in the filename.

@mohmdrezk @marcobarilari @CerenB #

## TODO

- [ ] show examples of filenames with the `run` and the `ses` entity
- [ ] break down the renaming tables by "workflow" to show how files are renamed after each analysis step

## func

| SPM                                                          | BIDS                                                        |
| ------------------------------------------------------------ | ----------------------------------------------------------- |
| s6usub-01_task-auditory_bold.nii                             | sub-01_task-auditory_space-individual_desc-smth6_bold.nii   |
| s6rsub-01_task-auditory_bold.nii                             | sub-01_task-auditory_space-individual_desc-smth6_bold.nii   |
| s6uasub-01_task-auditory_bold.nii                            | sub-01_task-auditory_space-individual_desc-smth6_bold.nii   |
| s6rasub-01_task-auditory_bold.nii                            | sub-01_task-auditory_space-individual_desc-smth6_bold.nii   |
| s6wusub-01_task-auditory_bold.nii                            | sub-01_task-auditory_space-IXI549Space_desc-smth6_bold.nii  |
| s6wrsub-01_task-auditory_bold.nii                            | sub-01_task-auditory_space-IXI549Space_desc-smth6_bold.nii  |
| s6wuasub-01_task-auditory_bold.nii                           | sub-01_task-auditory_space-IXI549Space_desc-smth6_bold.nii  |
| s6wrasub-01_task-auditory_bold.nii                           | sub-01_task-auditory_space-IXI549Space_desc-smth6_bold.nii  |
| s6sub-01_task-auditory_bold.nii                              | sub-01_task-auditory_desc-smth6_bold.nii                    |
| rp_sub-01_task-auditory_bold.txt                             | sub-01_task-auditory_desc-confounds_regressors.tsv          |
| rp_asub-01_task-auditory_bold.txt                            | sub-01_task-auditory_desc-confounds_regressors.tsv          |
| usub-01_task-auditory_bold.nii                               | sub-01_task-auditory_space-individual_desc-preproc_bold.nii |
| uasub-01_task-auditory_bold.nii                              | sub-01_task-auditory_space-individual_desc-preproc_bold.nii |
| rsub-01_task-auditory_bold.nii                               | sub-01_task-auditory_space-individual_desc-preproc_bold.nii |
| rasub-01_task-auditory_bold.nii                              | sub-01_task-auditory_space-individual_desc-preproc_bold.nii |
| std_usub-01_task-auditory_bold.nii                           | sub-01_task-auditory_space-individual_desc-std_bold.nii     |
| std_uasub-01_task-auditory_bold.nii                          | sub-01_task-auditory_space-individual_desc-std_bold.nii     |

## anat

| SPM                                                          | BIDS                                                        |
| ------------------------------------------------------------ | ----------------------------------------------------------- |
| wc1sub-01_T1w.nii                                            | sub-01_space-IXI549Space_res-bold_label-GM_probseg.nii      |
| wc2sub-01_T1w.nii                                            | sub-01_space-IXI549Space_res-bold_label-WM_probseg.nii      |
| wc3sub-01_T1w.nii                                            | sub-01_space-IXI549Space_res-bold_label-CSF_probseg.nii     |
| rc1sub-01_T1w.nii                                            | sub-01_space-individual_res-bold_label-GM_probseg.nii       |
| rc2sub-01_T1w.nii                                            | sub-01_space-individual_res-bold_label-WM_probseg.nii       |
| rc3sub-01_T1w.nii                                            | sub-01_space-individual_res-bold_label-CSF_probseg.nii      |
| wmsub-01_T1w.nii                                             | sub-01_space-IXI549Space_res-r1pt0_T1w.nii                  |
| wmsub-01_desc-skullstripped_T1w.nii                          | sub-01_space-IXI549Space_res-r1pt0_desc-preproc_T1w.nii     |
| msub-01_desc-skullstripped_T1w.nii                           | sub-01_space-individual_desc-preproc_T1w.nii                |
| wsub-01_desc-skullstripped_T1w.nii                           | sub-01_space-individual_res-r1pt0_desc-preproc_T1w.nii      |
| msub-01_space-individual_desc-something_label-brain_mask.nii | sub-01_space-individual_label-brain_mask.nii                |
| c1sub-01_T1w.surf.gii                                        | sub-01_desc-pialsurf_T1w.gii                                |

Not listed: some of the outputs of the segmentation done by the ALI toolbox for lesion detection

**Comment by Remi-Gau** [comment]
Note: show examples of filenames with the `run` and the `ses` entity.

**Comment by Remi-Gau** [comment]
Possibly helpful, adding some DAGs (directed acyclic graphs for the different workflows)

Without the workflow names
![segmentationSkullstripping](https://user-images.githubusercontent.com/6961185/148218128-860e9e27-af59-49cc-87f2-68e1ee03697b.png)

With the workflow names
![out](https://user-images.githubusercontent.com/6961185/148218337-b5270418-ec56-485f-9c6c-056d069498bb.png)

**Reply by marcobarilari** [reply]
These graphs are useful but at the same time hard to read so not sure if they can help in tracking the files before and after bids renaming.

**Reply by Remi-Gau** [reply]
I agree: they are too cluttered but in case someone wants to explore them

**Comment by marcobarilari** [comment]
So, don't know if still valid or too late but it looks good to me. I have played a lot with the `dev` branch of CPP_SPM and can say that the names are very clear and at least for my use case I am missing no name or any of them is creating confusion or problems.

**Comment by Remi-Gau** [comment]
Ideally I would like to add information in some json file explaining where each file came from because despite its non-bids character the spm prefix makes it very easy to know where the file came from (data provenance).
